src/backend/distributed/metadata/dependency.c

```diff
@@ -1218,12 +1218,14 @@ GetDependingView(Form_pg_depend pg_depend)
 	bool isMatView = get_rel_relkind(pg_rewrite->ev_class) == RELKIND_MATVIEW;
 	bool isDifferentThanRef = pg_rewrite->ev_class != pg_depend->refobjid;
 
-	systable_endscan(rscan);
-	relation_close(rewriteRel, AccessShareLock);
-
+	Oid dependingView = InvalidOid;
 	if ((isView || isMatView) && isDifferentThanRef)
 	{
-		return pg_rewrite->ev_class;
+		dependingView = pg_rewrite->ev_class;
```

Comment:
Don't we also need `HeapTupleIsValid(rewriteTup)` before doing all this?

Reply:
yeah, definitely, let me do that as well

Reply:
done